otelcol/collector.go

```diff
@@ -201,6 +209,16 @@ func (col *Collector) setupConfigurationComponents(ctx context.Context) error {
 	if err != nil {
 		return err
 	}
+	if col.updateConfigProviderLogger != nil {
+		col.updateConfigProviderLogger(col.service.Logger().Core())
+	}
+	if col.bc != nil {
+		x := col.bc.TakeLogs()
+		for _, log := range x {
+			ce := col.service.Logger().Core().Check(log.Entry, nil)
+			ce.Write(log.Context...)
```

Comment:
We should check `ce != nil` as done here: https://github.com/uber-go/zap/blob/fcf8ee58669e358bbd6460bef5c2ee7a53c0803a/logger.go#L246-L248

Reply:
That shouldn't be necessary, all the `CheckedEntry` methods are safe to call on nil: https://github.com/uber-go/zap/blob/fcf8ee58669e358bbd6460bef5c2ee7a53c0803a/zapcore/entry.go#L234-L236

Reply:
Hm, I doubt this will change but I am a little scared since this behavior is not guaranteed by the method's docstring